src/modules/iotjs_module_buffer.c

```diff
@@ -112,26 +112,26 @@ size_t iotjs_bufferwrap_length(iotjs_bufferwrap_t* bufferwrap) {
 }
 
 
-static size_t bound_range(int index, size_t low, size_t upper) {
-  if (index < (int)low) {
+static size_t bound_range(int32_t index, size_t low, size_t upper) {
```

Comment:
That was my first thought too. The problem with `size_t index` is on the call sites of `bound_range`. It is usually used without any validation after `JHANDLER_GET_ARG(1, number)` calls. So the first option is what I did in the PR and second one is to lot of copy-paste code to validate the number value before we call `bound_range`. I think this one is simpler and better maintainable.